File: DSAlgorithms/Leetcode/randoms/169-majorityElement.cpp
#include <iostream>
#include <vector>
int majorityElement(std::vector<int>& nums){
   int count = 0;
   int majorityElement = 0;
   for(int i = 0; i < nums.size(); i++)
   {
        if(count == 0)
            majorityElement = nums[i];
        if(majorityElement == nums[i])
            count++;
        else count--;
   }
   return majorityElement;
}

File: dsa_algo/leetcode/ts/src/old/169-majorityElement.ts
const majorityElement = (nums: number[]): number => {
    let count = 0;
    let majority_element = 0;
    let numsSize = nums.length;
    for(let element = 0;  element < numsSize; element++)
    {
        if(count === 0 )
            majority_element = nums[element];
        if(majority_element === nums[element])
            count++;
        else
            count--;
    }
    return majority_element;
}
